src/libs/DraftCommentUtils.ts

```diff
@@ -1,14 +1,14 @@
-import type {OnyxCollection, OnyxEntry} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
 import Onyx from 'react-native-onyx';
 import ONYXKEYS from '@src/ONYXKEYS';
+import type {DraftReportComments} from '@src/types/onyx';
 
-let draftCommentCollection: OnyxCollection<string> = {};
+let draftCommentCollection: OnyxEntry<DraftReportComments> = {};
 Onyx.connect({
```

Comment:
Wondering what it would take to nix this `Onyx.connect` usage? I'm looking at how it's used and it doesn't look like there's all that broad a surface, so it might be pretty simple to remove and use pure functions in this file instead

Reply:
I took a crack at it and I think it's out of scope for this PR. It ends up fanning out from `reasonForReportToBeInOptionList` to like _everything_. So far I've touched  25 files and I still have 79 outstanding type errors 😅. I'm not against seeing it through to the end, but it does dramatically increase the scope of features we need to test to get this PR merged